lotus/models/lm.py

```diff
@@ -1,12 +1,149 @@
-from abc import ABC, abstractmethod
+import os
+import functools
+import threading
+import ujson
 from typing import Any
+from abc import ABC, abstractmethod
+
+import litellm
+from litellm.caching import Cache
+
 
+litellm.cache = Cache(disk_cache_dir=".lotus_cache", type="disk")
 
-class LM(ABC):
+class LM:
     """Abstract class for language models."""
 
-    def _init__(self):
-        pass
+    def __init__(
+        self,
+        model: str,
+        model_type: str = 'chat',
+        temperature: float = 0.0,
+        cache: bool = True,
+        max_tokens: int = 512,
+        max_ctx_len: int = 4096,
+        **kwargs: dict[str, Any]
+    ):
+        self.model = model
+        self.model_type = model_type
+        self.cache = cache
+        self.max_ctx_len = max_ctx_len
+        self.kwargs = {
+            "temperature": temperature,
+            "max_tokens": max_tokens,
+            **kwargs
+        }
+        self.history = []
+
+    def __call__(
+        self, 
+        messages_batch: list | list[list], 
+        **kwargs: dict[str, Any]
+    ) -> list[str] | tuple[list[str], list[dict[str, Any]]]:
```

Comment:
Let's also add some types like `LMOutput` so it is more clear what is being returned. Check out `lotus/types.py` for some examples.